genesis/types.go

```diff
@@ -65,3 +94,71 @@ func GenesisDocFromJSON(jsonBlob []byte) (genState *GenesisDoc) {
 	}
 	return
 }
+
+//------------------------------------------------------------
+// Methods for genesis types
+// NOTE: breaks formatting convention
+// TODO: split each genesis type in its own file definition
+
+//------------------------------------------------------------
+// GenesisAccount methods
+
+// Clone clones the genesis account
+func (genesisAccount *GenesisAccount) Clone() GenesisAccount {
+	// clone the address
+	addressClone := make([]byte, len(genesisAccount.Address))
+	copy(addressClone, genesisAccount.Address)
+	// clone the account permissions
+	accountPermissionsClone := genesisAccount.Permissions.Clone()
+	return GenesisAccount{
+		Address:     addressClone,
+		Amount:      genesisAccount.Amount,
+		Name:        genesisAccount.Name,
+		Permissions: &accountPermissionsClone,
+	}
+}
+
+//------------------------------------------------------------
+// GenesisValidator methods
+
+// Clone clones the genesis validator
+func (genesisValidator *GenesisValidator) Clone() (GenesisValidator, error) {
+	if genesisValidator == nil {
+		return GenesisValidator{}, fmt.Errorf("Cannot clone nil GenesisValidator.")
+	}
+	if genesisValidator.PubKey == nil {
+		return GenesisValidator{}, fmt.Errorf("Invalid GenesisValidator %s with nil public key.",
+			genesisValidator.Name)
+	}
+	// clone the public key by writing and reading over go-wire serialisation
+	// TODO! write unit test to see whether this is correct
+	publicKeyClone, err := crypto.PubKeyFromBytes(genesisValidator.PubKey.Bytes())
```

Comment:
> because it allows you to make it explicit when you're copying data; and as a result avoid copying it when you don't have to.

You are copying data exactly when you don't have to because `PubKey.Bytes()` does a copy itself (as it ought to)